modules/ting_search/ting_search.module

```diff
@@ -444,24 +462,26 @@ function ting_search_search_execute($keys = NULL, $conditions = NULL) {
     }
   }
 
-  return $results;
+  return $results[$static_key];
 }
 
 /**
  * Retrive the result for a ting_search query performed in this request.
  *
+ * @param string $result_id
+ *   The search result id.
  * @param \Ting\Search\TingSearchResultInterface $new_search_result
  *   New search result to make globally accessible.
  *
  * @return \Ting\Search\TingSearchResultInterface|NULL
  *   The result or NULL if it is not available.
  */
-function ting_search_current_results(TingSearchResultInterface $new_search_result = NULL) {
-  $global_search_result = &drupal_static('ting_search_results');
+function ting_search_current_results($result_id = TING_SEARCH_DEFAULT_RESULT_ID, TingSearchResultInterface $new_search_result = NULL) {
```

Comment:
Given the latest changes I do not think result ids should be needed anymore? Everywhere it is referenced it now uses the default result id making this complexity unnecessary.